### PR TITLE
Start to sketch `internal/oci/remote.SignedImage[Index]`.

### PR DESCRIPTION
**Pull request by mattmoor**
This follows the pattern GGCR uses for functional `remote.Options`, and starts to define `SignedImage[Index]` methods, which build upon the `Signatures` implementation from a prior change.

This also tries to start adopting this pattern a few places, which drops bits of boilerplate (or changes them to functional options).

As part of this, I'm combining `FetchSignaturesForImage[Digest]` and tweaking it's signature a bit (no pun intended!), but I think long term we'll just want folks to consume the `remote` package directly themselves.

Signed-off-by: Matt Moore <mattomata@gmail.com>

#### Ticket Link

Related https://github.com/sigstore/cosign/issues/666

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by mattmoor** [comment]
Mostly opening this to run the automated tests, there are still several pieces that I want to add to this, and it needs unit test coverage for some of the new functionality in `internal/oci/remote`.

Please don't merge this, or scrutinize it too much yet 😅

**Comment by mattmoor** [comment]
Ok, I think I can explain all of the failures here with a thing I already had on my TODO list, which is nice.

tl;dr using `remote.SignedImage` on an image index is a bad idea 🤣 

I'll fix that after lunch.

File: internal/oci/remote/remote.go [hunk 1]
```diff
@@ -37,8 +38,148 @@ const (
...
+// signatures is a shared implementation of the oci.Signed* Signatures method.
+func signatures(digestable interface{ Digest() (v1.Hash, error) }, o *options) (oci.Signatures, error) {
+	h, err := digestable.Digest()
+	if err != nil {
+		return nil, err
+	}
+
+	// sha256:d34db33f -> sha256-d34db33f.suffix
+	tagStr := strings.ReplaceAll(h.String(), ":", "-") + o.signatureSuffix
+	ref := o.targetRepository.Tag(tagStr)
```
[the remark this answers is not in the document]
Comment (mattmoor) [reply internal/oci/remote/remote.go]: Pulled out `normalize`

File: internal/oci/remote/options_test.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+			AttestationSuffix: AttestationTagSuffix,
+			SBOMSuffix:        SBOMTagSuffix,
+			TargetRepository:  overrideRepo,
+			ROpt:              defaultOptions,
+		},
+	}, {
+		name: "signature option",
+		opts: []Option{WithSignatureSuffix(".pig")},
+		want: &options{
+			SignatureSuffix:   ".pig",
```
Comment (dekkagaijin) [inline internal/oci/remote/options_test.go:71]: ecdsa signatures end up being pretty small, actually

**Review by dekkagaijin — APPROVED** [review]
